src/productcatalogservice/server.go

```diff
@@ -147,13 +171,22 @@ func initProfiling(service, version string) {
 type productCatalog struct{}
 
 func parseCatalog() []*pb.Product {
-	var cat pb.ListProductsResponse
-
-	if err := jsonpb.Unmarshal(bytes.NewReader(catalogJSON), &cat); err != nil {
-		log.Warnf("failed to parse the catalog JSON: %v", err)
-		return nil
+	if reloadCatalog {
+		var catalog pb.ListProductsResponse
+		c, err := ioutil.ReadFile("products.json")
+		if err != nil {
+			log.Fatalf("failed to open product catalog json file: %v", err)
+			return nil
+		}
+		if err := jsonpb.Unmarshal(bytes.NewReader(c), &catalog); err != nil {
```

Comment:
this is duplicating the code above, including the file name. so it results in less maintainable code

can we have a method named `readCatalogFile()` referenced from both places instead?